ivr/src/main/java/org/motechproject/ivr/repository/CallDetailRecordDataService.java

```diff
@@ -25,4 +25,18 @@
     List<CallDetailRecord> findByProviderCallIdAndCallStatus(
             @LookupField(name = "providerCallId", customOperator = MATCHES_CASE_INSENSITIVE) String providerCallId,
             @LookupField(name = "callStatus", customOperator = MATCHES_CASE_INSENSITIVE) String callStatus);
+
+    @Lookup
+    List<CallDetailRecord> findByProviderCallIdWithCaseSensitive(@LookupField(name = "providerCallId") String providerCallId);
+
+    @Lookup
+    List<CallDetailRecord> findByCallStatus(@LookupField(name = "callStatus") String callStatus);
+
+    @Lookup
+    List<CallDetailRecord> findByMotechTimestamp(@LookupField(name = "motechTimestamp", customOperator = "LIKE") String motechTimestamp);
```

Comment:
I see similar solution in one motech classes so i assume that it should work, its not tested.